example-raft-kv/README.md

```diff
@@ -0,0 +1,36 @@
+# Example distributed KV store built upon openraft
+
+It is an example of how to build a real-world kv store with openraft.
+Includes:
+- An in-memory `RaftStorage` implementation [store](./store).
+
+- A server is based on [actix-web](https://docs.rs/actix-web/4.0.0-rc.2).  
+  Includes:
+  - raft-internal network APIs for replication and voting.
+  - Admin APIs to add nodes, change-membership etc.
+  - Application APIs to write a value by key or read a value by key.
+
+- Client and `RaftNetwork`([network](./network)) are built upon [reqwest](https://docs.rs/reqwest).
+
+## Run it
+
+```shell
+cargo build
+./example-raft-kv/test-cluster.sh
+```
+
+## Cluster management
+
+The raft itself does not store node addresses.
+But in a real-world application, the implementation of `RaftNetwork` needs to know the addresses.
```

Comment:
Question: I saw that rite_raft works by this way:

 - Start the leader informing the server address and raft address (aren't or could they be the same here?)
 - When adding a node, inform the server address, leader address and raft address.
 
 By doing that, the node will connect to the leader and append itself as an active instance. Could we do that in the demo? I am aware that you did that on `rpc 21001/write '{"AddNode":{"id":1,"addr":"127.0.0.1:21001"}}'` but is that better? Just a question.

Reply:
@drmingdrmer I can open a PR in your fork that includes this option, if the invocation includes the flag `leader_addr` it will auto include the leaner to it.

Reply:
Node addresses have to be stored and replicated, i.e., adding nodes has to be done through raft protocol.
Otherwise, a new leader won't know about the addresses of other nodes.

Adding a node as a member in a single step is can be done by just sending a `join(leader_address)` request to an uninitialized node and letting it do all these 3 steps: 
- Write a raft log entry to the leader to add the address of it: `rpc <leader>/write '{"AddNode":{"id":1,"addr":"127.0.0.1:21001"}}'`
- Inform the leader to replicate logs to it: `rpc <leader>/add-learner`.
- Inform the leader to change to a new membership config that includes it: `rpc <leader>/change-members`.